### DISCUSSION TITLE
Symlink multiple packages

### DISCUSSION THREAD

**Discussion by mfts** [pkgxdev/pkgx#364] [Help]
Asked via Discord by `DitchComfort`

Let say I want to actually use the GNU coreutils as default tools on my Mac, will they become defaults as long as I have the one-liner (magic) in my .zshrc? Or do I have to actually symlink every tool to e.g. /usr/local/bin for them to work?
I know Homebrew use a prefix g if you install coreutils, and you can choose to add a gnubin to the PATH of you want to use it without a prefix. Are there something similar if/when you install coreutils with tea?

I did try an example yesterday for dircolors and symlinked tea+dircolors to /usr/local/bin and everything was fine. But I really don’t want to symlink every tool, and what about man pages for those tools? Can I get the GNU man-pages?

**Comment by mxcl** [comment]
This is a good set of questions.

> Let say I want to actually use the GNU coreutils as default tools on my Mac, will they become defaults as long as I have the one-liner (magic) in my .zshrc? Or do I have to actually symlink every tool to e.g. /usr/local/bin for them to work?

They will be the defaults for you in your terminal. But scripts will not find them without the symlinks.

> I know Homebrew use a prefix g if you install coreutils, and you can choose to add a gnubin to the PATH of you want to use it without a prefix. Are there something similar if/when you install coreutils with tea?

Not yet, but we intend to have an “install” type solution at some point this year.

> I did try an example yesterday for dircolors and symlinked tea+dircolors to /usr/local/bin and everything was fine. But I really don’t want to symlink every tool

Yeah I feel you, I wouldn't want to do that either. The “install” type solution will be a real solution here

> and what about man pages for those tools? Can I get the GNU man-pages?

We add man pages to the env, but currently you would need to do `tea man foo`. We intend to add magic so tools like man will find everything in tea, but I'm still figuring out how that might work.

**Comment by DitchComfort** [comment]
So in the meantime I can probably symlink some of the tools I really want discovered everywhere...

I symlinked them all when I did the test the other day… tea+<name-of-tool>, so wasn't a big problem, but glad to hear there will be an easier "install"-type solution later this year (probably).

And thank you for the information regarding man pages, I didn't know that.

**Reply by mfts** [reply]
Hi @DitchComfort, it's possible to symlink to `~/.tea/local/bin` now. 

- related issue: https://github.com/orgs/teaxyz/discussions/600 
- pull request of feature: https://github.com/teaxyz/cli/pull/601

**Reply by DitchComfort** [reply]
Excellent 👌🏻 but I still don’t have the opportunity to use my Mac (or be online). But I’m really excited for all the new releases.
